src/docs/arch-backup.dox

```diff
@@ -0,0 +1,69 @@
+/*! @arch_page arch-backup Backup
+
+# Overview #
+
+Backup in WiredTiger can be performed using the \c "backup:" cursor. WiredTiger
+supports four types of backups.
+
+1. Full database
+2. Block-based incremental backup
+3. Log-based incremental backup
+4. Target backup
+
+The following page describes how backup works inside the WiredTiger. Please refer @ref backup
+for details on how to use each of these types of backups.
+
+# Full database #
+
+When the application opens the backup cursor, internally WiredTiger generates a list of all
+files in the database that are necessary for the backup. When generating that list WiredTiger
+takes both the checkpoint and schema locks to block database file modifications. There is a
+contract that all files that exist at the time the backup starts exist for the entire time
+the backup cursor is open. This contract applies to all files in the database, not just files
+that are included in the generated list.
```

Comment:
One thing that isn't clear is how long the checkpoint/schema locks are held - is it while generating the list or for the length of the backup?

Reply:
I reordered the sentence to give clarity on how long the locks are held.